### PR TITLE
Merge latest from main into dev for v2.7.0

### PR DESCRIPTION
**Pull request by cgillum**
Attempting to resolve a merge conflict between main and dev by merging the latest changes from main into dev.

### CODE REVIEW

**Comment by cgillum** [comment]
Looks like there were some packaging changes made directly to `main` that broken the .NET Isolated smoke test. Investigating.

**Comment by cgillum** [comment]
I was able to fix the .NET Isolated smoke test by redoing the SBOM packaging change to use the simplified method (the same one we're using for all the DTFx packages).

**Comment by cgillum** [comment]
dotnet restore step of publishing pipeline failed with this error:

```
       "D:\a\_work\1\s\test\DFPerfScenariosV1\DFPerfScenarios.csproj" (Restore target) (1) ->
       (Restore target) -> 
         D:\a\_work\1\s\test\DFPerfScenariosV1\DFPerfScenarios.csproj : error NU1107: Version conflict detected for Microsoft.Azure.WebJobs. Install/reference Microsoft.Azure.WebJobs 3.0.31 directly to project DFPerfScenarios to resolve this issue. 
       D:\a\_work\1\s\test\DFPerfScenariosV1\DFPerfScenarios.csproj : error NU1107:  DFPerfScenarios -> Microsoft.Azure.WebJobs.Extensions.DurableTask -> Microsoft.Azure.WebJobs (>= 3.0.31) 
       D:\a\_work\1\s\test\DFPerfScenariosV1\DFPerfScenarios.csproj : error NU1107:  DFPerfScenarios -> Microsoft.NET.Sdk.Functions 1.0.30 -> Microsoft.Azure.WebJobs (>= 2.3.0 && < 2.4.0).
```

Investigating.

**Comment by cgillum** [comment]
Found the issue. It looks like the release pipeline's "dotnet resotre" step covers more projects than what are included in our .sln. As a result, it's uncovering build problems that aren't covered by the CI. In this case, it's the DFPerfScenariosV1 project.

I'm resolving this by adding the project to our .sln and fixing the issue (changing the `net461` TFM to `net462`)

File: test/DFPerfScenariosV1/Tests/Counter.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ namespace DFPerfScenarios
     public static class CounterTest
 	{
 		[FunctionName("StartCounter")]
-		public static async Task<HttpResponseMessage> Start(
+		public static HttpResponseMessage Start(
```
Comment (cgillum) [inline test/DFPerfScenariosV1/Tests/Counter.cs:17]: Made this extra change to fix a build warning.

Comment (davidmrdavid) [reply test/DFPerfScenariosV1/Tests/Counter.cs:17]: Huh, interesting. It's not immediately clear to me why this change was needed, but I won't block on it

Comment (cgillum) [reply test/DFPerfScenariosV1/Tests/Counter.cs:17]: Not required - just cleaning up noise as I find issues getting the main objectives working.

**Review by davidmrdavid — APPROVED** [review]
CI's gone 🟢!

**Comment by cgillum** [comment]
Confirmed that the SBOM step worked too. :)

**Pull request merged**